.github/workflows/build.yml

```diff
@@ -149,37 +154,34 @@ jobs:
         find . -name "*.whl" -exec pip install {} \;
         pip install -r dev-requirements.txt
 
-    - name: Test Ubuntu
-      if: matrix.os == 'ubuntu-latest' && matrix.test-type == 'unit-tests'
+    - name: Test GUI
+      if: matrix.test-type == 'gui-tests'
       env:
         DISPLAY: ':99.0'
       run: |
         ci/github/start_herbstluftwm.sh &
         sleep 5
-        pushd tests
-        pytest ert_tests -sv --durations=0 -m "not integration_test"
+        pytest tests -sv --durations=5 --hypothesis-profile=ci -m "requires_window_manager"
+
+    - name: Test Ubuntu
+      if: matrix.os == 'ubuntu-latest' && matrix.test-type == 'unit-tests'
+      run: |
+        pytest tests -sv --durations=5 --hypothesis-profile=ci -m "not integration_test and not requires_window_manager"
 
     - name: Test MacOS
       if: matrix.os == 'macos-latest' && matrix.test-type == 'unit-tests'
       run: |
-        pushd tests
-        pytest ert_tests -sv --durations=0 -m "not integration_test"
+        pytest tests -sv --durations=5 --hypothesis-profile=ci -m "not integration_test and not requires_window_manager"
```

Comment:
"Test MacOS" is identical to "Test Ubuntu". Combine them?

Reply:
Will do